ui/components/app/snaps/snap-metadata-modal/snap-metadata-modal.js

```diff
@@ -0,0 +1,245 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+import { getSnapPrefix, stripSnapPrefix } from '@metamask/snaps-utils';
+import {
+  getSnap,
+  getSnapRegistryData,
+  getTargetSubjectMetadata,
+} from '../../../../selectors';
+import {
+  Box,
+  ButtonLink,
+  Modal,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+} from '../../../component-library';
+import {
+  AlignItems,
+  Display,
+  FlexDirection,
+  FlexWrap,
+  FontWeight,
+  JustifyContent,
+  OverflowWrap,
+  TextAlign,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import SnapAvatar from '../snap-avatar';
+import { formatDate, getSnapName } from '../../../../helpers/utils/util';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { getPhishingResult } from '../../../../store/actions';
+import { useOriginMetadata } from '../../../../hooks/useOriginMetadata';
+import { SnapDelineator } from '../snap-delineator';
+import { DelineatorType } from '../../../../helpers/constants/snaps';
+import { ShowMore } from '../show-more';
+import SnapExternalPill from '../snap-version/snap-external-pill';
+import InfoTooltip from '../../../ui/info-tooltip';
+
+export const SnapMetadataModal = ({ snapId, isOpen, onClose }) => {
+  const t = useI18nContext();
+
+  const subjectMetadata = useSelector((state) =>
+    getTargetSubjectMetadata(state, snapId),
+  );
+
+  const snap = useSelector((state) => getSnap(state, snapId));
+
+  const versionHistory = snap?.versionHistory ?? [];
+  const installInfo = versionHistory.length
+    ? versionHistory[versionHistory.length - 1]
+    : undefined;
+
+  const installOrigin = useOriginMetadata(installInfo?.origin);
+
+  const snapName = getSnapName(snapId, subjectMetadata);
+  const snapPrefix = getSnapPrefix(snapId);
+  const packageName = stripSnapPrefix(snapId);
+  const isNPM = snapPrefix === 'npm:';
+  const versionPath = subjectMetadata?.version
+    ? `/v/${subjectMetadata?.version}`
+    : '';
+  const url = isNPM
+    ? `https://www.npmjs.com/package/${packageName}${versionPath}`
+    : packageName;
+
+  const snapRegistryData = useSelector((state) =>
+    getSnapRegistryData(state, snapId),
+  );
+
+  const { website = undefined } = snapRegistryData?.metadata ?? {};
+  const [safeWebsite, setSafeWebsite] = useState(null);
+
+  useEffect(() => {
+    const performPhishingCheck = async () => {
+      const phishingResult = await getPhishingResult(website);
+
+      if (!phishingResult.result) {
+        setSafeWebsite(new URL(website));
```

Comment:
Can we move this to a shared hook between the modal and snaps settings?

Feels like there is a lot of duplication between those two components?

Reply:
implemented in cd5734f